### PR TITLE
add testing with ats

### PR DESCRIPTION
**Pull request by mleisa**
This PR:

- fixes `.abs/main.yaml`
- adds ats

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Make sure `values.yaml` and `values.schema.json` are valid.

### CODE REVIEW

**Comment by mleisa** [comment]
the `values.yaml` / `values.schema.json` test is failing atm, though I haven't changed the two files. Could it be that the test is counting the `tests/test-values.yaml` file that I have added as such a change? Because the test checks against the string "values.yaml" which technically exists in `test-values.yaml`

File: .abs/main.yaml [hunk 1]
```diff
@@ -2,3 +2,5 @@ replace-chart-version-with-git: true
 generate-metadata: true
 chart-dir: ./helm/cloudflared/
 destination: ./build
+# CI overwrites this, check .circleci/config.yaml
+catalog-base-url: https://giantswarm.github.io/giantswarm-catalog/
```
Comment (Gacko) [inline .abs/main.yaml:6]:
```suggestion
catalog-base-url: https://giantswarm.github.io/giantswarm-catalog/
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -12,7 +12,9 @@ workflows:
           app_catalog: "giantswarm-catalog"
           app_catalog_test: "giantswarm-test-catalog"
           chart: "cloudflared"
+          persist_chart_archive: true
           # Trigger job on git tag.
           filters:
             tags:
               only: /^v.*/
+          push_to_oci_registry: true
```
Comment (Gacko) [inline .circleci/config.yml:20]:
```suggestion
          push_to_oci_registry: true
```

**Comment by ubergesundheit** [comment]
> the `values.yaml` / `values.schema.json` test is failing atm, though I haven't changed the two files. Could it be that the test is counting the `tests/test-values.yaml` file that I have added as such a change? Because the test checks against the string "values.yaml" which technically exists in `test-values.yaml`

Please update your branch to the lastest master, I think the required changes were introduced by https://github.com/giantswarm/cloudflared-app/pull/28/files#diff-c70a68e380af0b06885a3e05a8062ded335ec477479b13ebecfdc9f016d7b97d

File: .ats/main.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+app-tests-app-config-file: tests/test-values.yaml
+
+skip-steps: functional
+
+smoke-tests-cluster-type: kind
+
+upgrade-tests-cluster-type: kind
+upgrade-tests-app-catalog-url: https://giantswarm.github.io/giantswarm-catalog
+upgrade-tests-app-config-file: tests/test-values.yaml
```
Comment (ubergesundheit) [inline .ats/main.yaml:9]: Why is this not used for smoke tests?

**Comment by ubergesundheit** [comment]
In case you want to run tests also in CI, you'll need to add some lines to the circleCI config file (please see other projects for reference, cert-manager for example)

File: .ats/main.yaml (hunk 3 above)
Comment (mleisa) [reply .ats/main.yaml:9]: Not sure what you mean. it defines kind for smoke test clusters in line 5, or do you mean something else?

Comment (ubergesundheit) [reply .ats/main.yaml:9]: the `upgrade-tests-app-config-file: tests/test-values.yaml` line. Why is the test config file not used for smoke tests? Its possible to specify analogue to this using `smoke-tests-cluster-config-file`

File: .circleci/config.yml [hunk 4]
```diff
@@ -18,3 +18,13 @@ workflows:
...
               only: /^v.*/
           push_to_oci_registry: true
+
+      - architect/run-tests-with-ats:
+          name: execute chart tests
+          filters:
+            # Do not trigger the job on merge to master.
+            branches:
+              ignore:
+                - master
```
Comment (ubergesundheit) [inline .circleci/config.yml:28]:
```suggestion
                - main
```

**Review by ubergesundheit — CHANGES_REQUESTED** [review]

File: .ats/main.yaml (hunk 3 above)
Comment (ubergesundheit) [reply .ats/main.yaml:9]: Scratch that, its already good like it is

**Comment by Gacko** [comment]
Is this still ongoing? 🧐

**Comment by Gacko** [comment]
Reviewed this just to get it removed from my "Review requested" list.

**Review by Gacko — APPROVED** [review]

**Pull request closed without merging**